### PR TITLE
Do not load CAN log into memory when replaying

### PR DESCRIPTION
**Pull request by sophokles73**
Currently, a CAN dump file is being loaded into memory before starting to replay its frames to the databroker.
This PR changes the behavior to read the CAN frames from the dump file one-by-one instead of pre-loading the whole file into memory.

### CODE REVIEW

File: dbc2val/dbcfeederlib/canplayer.py [hunk 1]
```diff
@@ -40,73 +38,42 @@ class CANplayer:
...
+                log.debug("Failed to send message via CAN bus")
 
-    def getNextMessage(self):
-        val = self.messages[self.indexout]
-        self.indexout += 1
-        # in case of end of array set the index pointer to start again
-        if self.indexout >= self.index:
-            self.indexout = 1
-        return val
+        log.debug("Replayed all messages from CAN log file")
```
Comment (erikbosch) [inline dbc2val/dbcfeederlib/canplayer.py:61]: I think this could be a log.info, can be quite useful and shall only be given once, right?

Comment (sophokles73) [reply dbc2val/dbcfeederlib/canplayer.py:61]: this is logged each time the log has been re-played fully.

File: dbc2val/dbcfeederlib/canplayer.py [hunk 2]
```diff
@@ -40,73 +38,42 @@ class CANplayer:
...
-        txThread = threading.Thread(target=self.txWorker)
-        txThread.start()
+            if not self.run:
+                break
+            try:
+                self.bus.send(msg)
+                log.debug(f"Message sent on {self.bus.channel_info}")
+                log.debug(f"Message: {msg}")
+            except can.CanError:
+                log.debug("Failed to send message via CAN bus")
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply dbc2val/dbcfeederlib/canplayer.py:59]: no particular reason, I tend to keep log statements at a low level to reduce verbosity. However, in this case it probably makes sense to raise the level accordingly. We expect this to happen very seldomly only, I guess.

**Comment by sophokles73** [comment]
@erikbosch I have adjusted the log levels

File: dbc2val/dbcfeederlib/canplayer.py (hunk 1 above)
Comment (erikbosch) [reply dbc2val/dbcfeederlib/canplayer.py]: Ah, then it can actually come quite often if we run the log repeatedely

**Review by erikbosch — APPROVED** [review]
Did a sanity test (native together with databroker) and it worked without problem, ok for me

**Review by SebastianSchildt — APPROVED** [review]
Nice. I wanted this for a long time, never came around to it. Now even the the huge logs starts instantaneously. And less lines. Seems I can have my 🍰  and eat it too

**Pull request merged**